packages/forms/src/elements/common/Label.tsx

```diff
@@ -71,6 +71,32 @@ export const Label = React.forwardRef<HTMLLabelElement, ILabelProps>((props, ref
       </StyledRadioLabel>
     );
   } else if (type === 'checkbox') {
+    /**
+     * `onLabelSelect` is a workaround for checkbox label `shift + click` bug in Firefox
+     * See: https://bugzilla.mozilla.org/show_bug.cgi?id=559506
+     */
+    const onLabelSelect = (e: React.KeyboardEvent) => {
+      const isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+
+      if (fieldContext && isFirefox) {
+        const inputProps = fieldContext.getInputProps();
+        const input = document.getElementById(inputProps.id as string) as HTMLInputElement;
+
+        if (input && input.type === 'checkbox') {
+          if (e.shiftKey) {
+            input.click();
+            input.checked = true;
```

Comment:
How does this work when an element is controlled?

Reply:
It _seems_ to hold up: https://5f3c3e29d6b3cf01999d887e--zendeskgarden-react-components.netlify.app/forms/#advanced

Reply:
Although, the prop for `indeterminate` doesn't seem to be getting respected here.

Reply:
I'll need to consider the implications of a controlled checkbox. I believe there may need to be some changes. Please hold 👍  .

Reply:
Currently, controlled usages do not work in Firefox. The controlled example should have the ability to have the checkbox set to indeterminate, _but this does not work_. Thanks for catching this @austin94 🙏 . 

My new (slightly modified) proposal: only address the Firefox-specific bug in _uncontrolled_ checkboxes. I could add an additional boolean check (e.g `isControlled`) and only apply the fix for _uncontrolled_ checkboxes. The big win here is that the selectable `Table` components will receive the fix and controlled checkboxes work as they did before.

New commit coming shortly.

Reply:
**Edit**: I went back and tried various controlled examples (custom set checked buttons, etc). It looks like when a checkbox is controlled, the controlled state will be applied to the checkbox. Here's how I verified this:

I added the following to the controlled example:

```jsx
<button
  onClick={() => {
    setState({
      checkboxChecked: true,
      checkboxIndeterminate: false
    });
  }}
>
  Set checked
</button>
<button
  onClick={() => {
    setState({
      checkboxIndeterminate: true
    });
  }}
>
  Set indeterminate
</button>
```

Then I performed `shift + click` the checkbox label to check and uncheck the checkbox. Then to verify that controlled usages worked, I clicked the "Set checked" and "Set indeterminate" buttons.

![2020-08-18 18 17 20](https://user-images.githubusercontent.com/1811365/90581052-2c62f380-e17f-11ea-8aaf-75311044bcf3.gif)

@austin94, do you see any issues regarding controlled checkboxes?

Reply:
> Although, the prop for indeterminate doesn't seem to be getting respected here.

I noticed this too during the controlled checkbox usage investigation. It seems like indeterminate wasn't respected prior to this PR.